.platform.yaml

```diff
@@ -1,4 +1,4 @@
-required_version: =~ 3
+required_version: 3.2.2.0
```

Comment:
I'm glad you did this because it did expose the bug (`setup-platform-action` needs a new release since being updated to default to `3.2.2.0`, which I'll go and fix now), but I don't think you want to leave it like this all the time.

Requiring a _specific_ version is very restricted. Every time we ship a new version and change `setup-platform-action`s default, your next CI will fail and you'll have to do something. If you really do want to do this, you'll want to also update the workflow to specify this same version instead of relying on the ever-changing default.

But even if you do that, this requirement will still be enforced locally too, and folks across the team having similar-but-not-the-same PlatformCLI versions is pretty common and acceptable, but now this hard requirement will error on anyone not using this _exact_ version. As we make use of PlatformCLI more and more, that'll start being more and more of a problem.

That's why `=~` is good, because it can require a certain minimum version and allow newer than those, within reason. This is to avoid breakage on (e.g.) major version bumps -- just like you'd approach any app dependency.

The hope is teams understand this and decide to only be as restrictive as they need. This particular app is not super critical, right? You may not want to be very restrictive at all. That's how `=~ 3` ended up here to begin with, but maybe `=~ 3.2` is better, I'm not sure. But I'm _pretty_ sure a full and exact `=` constraint is too restrictive.

WDYT?

Reply:
Totally agree.  Pinning to a hyper-specific version can lead to pain down the line.  My other comment was just around how to get this deployed as fast as possible.  I should have been more clear about that context

Reply:
yeah, my preference is to be the least restrictive possible... just made this change to force the update, but now I see it might not be needed. I'll go back to using =~ 3

Reply:
Cool, yeah sorry if I wasn't clear -- the key thing was to look at the build logs to see what actually happened, since (if we ignore the fact that I missed a step) we didn't expect to have to _do_ anything.

For example, you can see what version it installed now, and it's correct:

![](https://files.pbrisbin.com/screenshots/screenshot.1497749.png)

Then further down we can see it making a change to the `Bucket`:

![](https://files.pbrisbin.com/screenshots/screenshot.1498139.png)

This is pretty good evidence it's doing what we want. After it's done, we can ask someone to look actually in AWS to see that the lifecycle is gone, if we want.